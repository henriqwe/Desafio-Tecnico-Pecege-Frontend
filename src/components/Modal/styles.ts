import styled from 'styled-components'
import * as Icons from '@heroicons/react/24/outline'

export const CheckIcon = styled(Icons.CheckIcon)`
  width: 1rem;
  height: 1rem;
`
export const XMarkIcon = styled(Icons.XMarkIcon)`
  width: 1rem;
  height: 1rem;
`
