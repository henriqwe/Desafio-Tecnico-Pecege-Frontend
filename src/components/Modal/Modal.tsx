import * as AlertDialog from '@radix-ui/react-alert-dialog'
import './styles.css'
import * as S from './styles'
import { ReactNode } from 'react'
import { Button } from '&components/Button/Button'

interface props {
  triggerContent?: ReactNode
  action: () => void
  title: string
  content?: ReactNode
  cancelText?: string
  actionsText?: string
  open?: boolean
  onOpenChange?: React.Dispatch<React.SetStateAction<boolean>>
}
export function Modal({
  triggerContent,
  action,
  title,
  content,
  open,
  onOpenChange,
  cancelText = 'NÃ£o',
  actionsText = 'Sim'
}: props) {
  return (
    <AlertDialog.Root open={open} onOpenChange={onOpenChange}>
      {triggerContent && (
        <AlertDialog.Trigger asChild>
          <button>{triggerContent}</button>
        </AlertDialog.Trigger>
      )}
      <AlertDialog.Portal>
        <AlertDialog.Overlay className="AlertDialogOverlay" />
        <AlertDialog.Content className="AlertDialogContent">
          <AlertDialog.Title className="AlertDialogTitle">
            {title}
          </AlertDialog.Title>
          <AlertDialog.Description className="AlertDialogDescription">
            {content}
          </AlertDialog.Description>
          <S.AlertWrapper>
            <AlertDialog.Cancel asChild>
              <div>
                <Button variant="danger">
                  <span>{cancelText}</span> <S.XMarkIcon />
                </Button>
              </div>
            </AlertDialog.Cancel>
            <AlertDialog.Action asChild>
              <Button variant="primary" onClick={() => action()}>
                <span>{actionsText}</span> <S.CheckIcon />
              </Button>
            </AlertDialog.Action>
          </S.AlertWrapper>
        </AlertDialog.Content>
      </AlertDialog.Portal>
    </AlertDialog.Root>
  )
}
