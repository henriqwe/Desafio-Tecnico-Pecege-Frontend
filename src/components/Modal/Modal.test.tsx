import { fireEvent, render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Modal } from './Modal'

const fnMoked = jest.fn()

describe('Modal', () => {
  it('should render correctly snapshot', async () => {
    const { container, findByTestId } = render(
      <Modal
        action={fnMoked}
        title="modal test"
        triggerContent={<div>trigger</div>}
      />
    )

    const modalTrigger = await findByTestId('modal-trigger')

    await waitFor(async () => fireEvent.click(modalTrigger))

    await waitFor(async () => findByTestId('modal-button-ok'))

    const modalButtonOk = await findByTestId('modal-button-ok')

    waitFor(async () => fireEvent.click(modalButtonOk))

    expect(container).toMatchSnapshot()
    expect(fnMoked).toBeCalled()
  })
})
