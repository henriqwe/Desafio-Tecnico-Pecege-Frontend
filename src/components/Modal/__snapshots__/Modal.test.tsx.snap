// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly snapshot 1`] = `
<div>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    data-testid="modal-trigger"
    type="button"
  >
    <div>
      trigger
    </div>
  </button>
</div>
`;
